src/cargo.rs

```diff
@@ -0,0 +1,140 @@
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+use std::io::prelude::*;
+use err::ProfError;
+use regex::Regex;
+use std::process::{Command, exit};
+use std::path::Path;
+
+/// Returns the closest ancestor path containing a `Cargo.toml`.
+///
+/// Returns `None` if no ancestor path contains a `Cargo.toml`, or if
+/// the limit of MAX_ANCESTORS ancestors has been reached.
+pub fn find_toml() -> Option<PathBuf> {
```

Comment:
I don't think searching for a `target` dir is 100% reliable, you'll need to account for things like: https://github.com/rust-lang/cargo/pull/1657 

I think using cargo as a library might be the most reliable way to ensure things work reliably, you could inspect how `cargo read-manifest` is written perhaps as a starting point (or another `cargo-foo` package?).